server/utils/security/output_redaction.py

```
@@ -0,0 +1,192 @@
+            ))
+    findings.sort(key=lambda f: (f.start, f.end))
+    return findings
+
+
+def scan(text: str) -> list[Finding]:
+    """Return non-overlapping findings sorted by position. Never raises."""
+    if not text:
+        return []
+    try:
+        return _scan_unsafe(text)
+    except Exception:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)